sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -556,10 +581,233 @@ public String toJson() {
+        Map<String, ClientSideRequestStatistics.AddressResolutionStatistics> addressResolutionStatisticsMap
+    ) {
+        if (addressResolutionStatisticsMap == null || addressResolutionStatisticsMap.size() == 0) {
+            return;
+        }
+
+        for (ClientSideRequestStatistics.AddressResolutionStatistics addressResolutionStatistics
+            : addressResolutionStatisticsMap.values()) {
+
+            if (addressResolutionStatistics.isInflightRequest() ||
+                addressResolutionStatistics.getEndTimeUTC() == null) {
+
+                // skipping inflight or failed address resolution statistics
+                // capturing error count etc. won't make sense here - request diagnostic
+                // logs are the right way to debug those - not metrics
+                continue;
+            }
+
+            Duration latency = Duration.between(
+                addressResolutionStatistics.getStartTimeUTC(),
+                addressResolutionStatistics.getEndTimeUTC());
+
+            StringBuilder sb = new StringBuilder();
+            sb.append("AddressResolution|");
+            sb.append(addressResolutionStatistics.getTargetEndpoint());
+            sb.append("|");
+            if (addressResolutionStatistics.isForceRefresh()) {
+                sb.append("1|");
+            } else {
+                sb.append("0|");
+            }
+
+            if (addressResolutionStatistics.isForceCollectionRoutingMapRefresh()) {
+                sb.append("1");
+            } else {
+                sb.append("0");
+            }
+
+            CosmosDiagnosticsRequestInfo info = new CosmosDiagnosticsRequestInfo(
+                requestStats.getActivityId(),
+                null,
+                null,
+                sb.toString(),
+                addressResolutionStatistics.getStartTimeUTC(),
+                latency,
+                null,
+                0,
+                0,
+                0,
+                0,
+                new ArrayList<>()
+            );
+
+            requestInfo.add(info);
+        }
+    }
+
+    /**
+     * Gets a collection of {@link CosmosDiagnosticsRequestInfo} records providing more information about
+     * individual requests issued in the transport layer to process this operation.
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
+     * a custom {@link CosmosDiagnosticsHandler}
+     * @return a collection of {@link CosmosDiagnosticsRequestInfo} records providing more information about
+     * individual requests issued in the transport layer to process this operation.
+     */
+    public Collection<CosmosDiagnosticsRequestInfo> getRequestInfo() {
```

Comment:
How about `List` in place of `Collection`?

Reply:
This is public API - I don't think we need/should guarantee indexed access. Ordering and being to iterate over it should be sufficient. Also consistent in how we exposed getDiagnostics for example.